pandas/tests/groupby/test_groupby.py

```diff
@@ -2348,6 +2348,20 @@ def test_groupby_duplicate_index():
     tm.assert_series_equal(result, expected)
 
 
+def test_group_on_empty_multiindex(transformation_func):
+    # GH 47787
+    # With one row, those are transforms so the schema should be the same
+    df = DataFrame(data=[[1, 2, 3, 4]], columns=["col_1", "col_2", "col_3", "col_4"])
+    df = df.set_index(["col_1", "col_2"])
+    result = df.groupby(["col_1"]).fillna("")
+    assert df.index.names == result.index.names
```

Comment:
Could you use `tm.assert_frame_equal(result, expected)` where `expected` is the expected resulting DataFrame? Index objects and it's attributes should be compared as well.

Reply:
There was an issue in my test, I had hardcoded fillna whereas I wanted to test all functions - in taht case, for all the cum functions, the schema is actually different (returns a series). I could split those cases but that is a bit more fundamental than what I was after.

Reply:
It's the projects preferred testing style if you could use one of the `tm.assert_*_equal` functions as testing the objects is good too.

Reply:
@mroeschke fixed now - we're using the `tm.assert_index_equals`